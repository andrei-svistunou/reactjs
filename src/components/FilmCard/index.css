.film__item {
  display: flex;
  margin-top: 30px;
}

.film__poster {
  width: 30%;
}

.film__info {
  width: 70%;
  padding-left: 40px;
}

.film__img {
  width: 100%;
}

.film-title {
  color: #fc7e8d;
  font-size: 28px;
  margin-top: 10px;
}

.film__rating {
  color: yellowgreen;
  border: 1px solid yellowgreen; 
  border-radius: 50%;
  padding: 7px;
  font-size: 20px;
  font-weight: 400;
  margin-left: 10px;
}

.film__tagline,
.film__year,
.film__duration,
.film__description {
  color: #fff;
}

.film__duration {
  margin-left: 20px;
}

.film__year,
.film__duration {
  font-weight: 600;
}

.row {
  padding: 10px 0;
}

.film__tagline,
.film__description {
  opacity: .7;
}