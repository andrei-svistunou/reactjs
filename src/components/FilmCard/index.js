import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import getMovie from '../../redux/actions/getMovie';
import './index.css';

class FilmCard extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    tagline: PropTypes.string.isRequired,
    release_date: PropTypes.string.isRequired,
    genres: PropTypes.array.isRequired,
    poster_path: PropTypes.string.isRequired,
    overview: PropTypes.string.isRequired,
    runtime: PropTypes.number.isRequired,
    vote_average: PropTypes.number.isRequired,
    getMovie: PropTypes.func.isRequired
  }

  componentDidMount() {
    const { getMovie } = this.props;

    getMovie(this.props.match.params.id);
  }

  render() {
    const {
      poster_path,
      title,
      vote_average,
      tagline,
      release_date,
      runtime,
      overview
    } = this.props;

    return (
    <section className='film__item'>
      <div className="film__poster">
        <img className="film__img" src={poster_path}/>
      </div>
      <div className="film__info">
        <h3 className="film-title">{title}
          <span className="film__rating">{vote_average}</span>
        </h3>
        <p className="row film__tagline">{tagline}</p>
        <p className="row">
          <span className="film__year">{release_date}</span>
          <span className="film__duration">{runtime} min</span>
        </p>
        <p className="row film__description">{overview}</p>
      </div>
    </section>
    );
  }
}

const mapStateToProps = state => state.films.selectedFilm;

const mapDispatchToProps = (dispatch) => {
  const actions = bindActionCreators({
    getMovie
  }, dispatch);
  return { ...actions };
};

export default connect(mapStateToProps, mapDispatchToProps)(FilmCard);
