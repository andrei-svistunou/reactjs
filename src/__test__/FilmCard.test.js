import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import FilmCard from '../components/FilmCard';

describe('<FilmCard />', () => {
  const filmProps = {
    title: 'Best film',
    release_date: '1982',
    genres: ['Drama', 'Adventure'],
    poster_path: 'https://google.com/img/3',
    vote_average: 8
  };

  test('should render a FilmCard', () => {
    const filmCard = shallow(<FilmCard film={filmProps}/>);

    expect(toJson(filmCard)).toMatchSnapshot();
  });
});
