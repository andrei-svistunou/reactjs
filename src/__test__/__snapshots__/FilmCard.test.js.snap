// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmCard /> should render a FilmCard 1`] = `
<section
  className="film__item"
>
  <div
    className="film__poster"
  >
    <img
      className="film__img"
      src="https://google.com/img/3"
    />
  </div>
  <div
    className="film__info"
  >
    <h3
      className="film-title"
    >
      Best film
      <span
        className="film__rating"
      >
        8
      </span>
    </h3>
    <p
      className="row film__tagline"
    />
    <p
      className="row"
    >
      <span
        className="film__year"
      >
        1982
      </span>
      <span
        className="film__duration"
      >
         min
      </span>
    </p>
    <p
      className="row film__description"
    />
  </div>
</section>
`;
